packages/EDA/src/pls/pls-ml.ts

```diff
@@ -0,0 +1,378 @@
+// Predicitve tools based on the PLS method
+
+import * as grok from 'datagrok-api/grok';
+import * as ui from 'datagrok-api/ui';
+import * as DG from 'datagrok-api/dg';
+
+import {TITLE, RESULT_NAMES} from './pls-constants';
+import {getPlsAnalysis, PlsOutput, getLines} from './pls-tools';
+import {LINK} from './pls-constants';
+import {getPredictionByLinearRegression} from '../regression';
+
+// PLS ML specific constants
+const EXTRA_ROWS = 1;
+const SHIFT = 2;
+const MIN_LOADINGS = 1;
+const MIN_COLS_COUNT = SHIFT + MIN_LOADINGS;
+const SIZE_ARR_LEN = 2;
+const MODEL_IDX = 0;
+const SCORES_IDX = 1;
+const BYTES_PER_SIZES = SIZE_ARR_LEN * 4;
+const BLOCK_SIZE = 64;
+
+/** Titles */
+/*enum TITLE {
+  FEATURES = 'Feature names',
+  PARAMS = 'Regression Coefficients',
+  X_LOADING = 'x.loading.p',
+  LOADINGS = 'Loadings',
+  COMP = 'component',
+  COMPS = 'components',
+  EXPL_VAR = 'explained variance',
+  X_SCORE = 'x.score.t',
+  Y_SCORE = 'y.score.u',
+  SCORES = 'Scores',
+};*/
```

Comment:
If we dont need this, i'd rather not push it

Reply:
Removed!